pkg/pagination/pagination.go

```diff
@@ -50,6 +75,25 @@ type PagedResponse[T any] struct {
 	Items      []T  `json:"items"`
 }
 
+// NewPagedResponse creates a new PagedResponse with the given page, totalCount and items.
+func NewPagedResponseWith[OUT any, IN any](resp PagedResponse[IN], m func(in IN) (OUT, error)) (PagedResponse[OUT], error) {
```

Comment:
I'm not convinced we need this. e.g.

```go
apiRes := pagination.PagedResponse{
  Page: res.Page,
  TotalCount: res.TotalCount,
  Items: slicesx.Map(...)
}
```

looks fine to me...

Reply:
The challenge is if that then we have to copy struct properties everywhere in router which is easy to forget + error prone if we add new properties to PagedResponse like hasNextPage